### PR TITLE
364 sharing photos to inat

### PR DESCRIPTION
**Pull request by budowski**

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -31,6 +42,73 @@ const App = ( { children }: Props ): Node => {
...
+
+  useEffect( () => {
+    const listener = ShareMenu.addNewShareListener( handleShare );
+
+    return () => {
+      listener.remove();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [] );
+
```
Comment (kueda) [inline src/components/App.js:111]: This is a lot to pack into our root component. Could we move this into a hook?

**Review by kueda — CHANGES_REQUESTED** [review]
Things seem to be working in Android. For iOS, when I run pod install I get this:

```
[!] Unable to determine whether to build `React-jsi` as a module due to a conflict between the following target definitions:
	- `Pods-iNaturalistReactNative` requires `React-jsi` as a module
	- `Pods-iNaturalistReactNative-ShareExtension` does not require `React-jsi` as a module

Defaulting to skip building `React-jsi` as a module.
```

When I try to run in a Simulator the build fails with this:

```
2023-04-17 17:45:36.275 xcodebuild[98474:50833906]  DVTAssertions: Warning in /System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot2/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-21506/DTDeviceKitBase/DTDKRemoteDeviceData.m:382
Details:  (null) deviceType from 00008103-000A44C23AE9401E was NULL when -platform called.
Object:   <DTDKMobileDeviceToken: 0x123828480>
Method:   -platform
Thread:   <NSThread: 0x60000083d0c0>{number = 4, name = (null)}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
** BUILD FAILED **


The following build commands failed:
	SwiftEmitModule normal arm64 Emitting\ module\ for\ iNaturalistReactNative_ShareExtension (in target 'iNaturalistReactNative-ShareExtension' from project 'iNaturalistReactNative')
(1 failure)
```

Any ideas?

**Comment by kueda** [comment]
Ok, after the recent changes, I can get this to build and work in Android and iOS, and it seems to work as expected, so remaining todos are

- [ ] pack the App.js config stuff away into a hook or some other file
- [ ] resolve conflicts
- [ ] let everyone on the team know that they may need to run this in xcode at least once before it will build properly for iOS

**Review by kueda — CHANGES_REQUESTED** [review]
Aside from still needing to resolve conflicts, when I run this on a device and share a photo, I get this exception:

```
RangeError: Invalid time value
    at formatISO (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:126147:27)
    at formatISONoTimezone (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:119521:85)
    at formatExifDateAsString (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:358532:85)
    at ?anon_0_ (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:358143:113)
    at next (native)
    at asyncGeneratorStep (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:1232:26)
```

When I try to run on an iOS simulator, I get this build error:

```
<unknown>:0: error: module map file '/Users/kueda/Library/Developer/Xcode/DerivedData/iNaturalistReactNative-bodcddsbdubblxajopponrkjzdoy/Build/Products/Debug-iphonesimulator/RNShareMenu/RNShareMenu.modulemap' not found
/Users/kueda/projects/iNaturalistReactNative/node_modules/react-native-share-menu/ios/ShareViewController.swift:15:8: error: no such module 'RNShareMenu'
import RNShareMenu
       ^

SwiftCompile normal arm64 Compiling\ ShareViewController.swift /Users/kueda/projects/iNaturalistReactNative/node_modules/react-native-share-menu/ios/ShareViewController.swift (in target 'iNaturalistReactNative-ShareExtension' from project 'iNaturalistReactNative')
    cd /Users/kueda/projects/iNaturalistReactNative/ios
```

Opening in xcode and running from there does not seem to address the problem like it did before.

**Comment by budowski** [comment]
@kueda -
1. Pushed a fix for the iOS simulator issue
2. However, couldn't recreate the crash you've experienced - tried iOS device, iOS simulator and Android. I did add an additional check there that *might* prevent the crash + some additional debug print
3. Something to note - while trying to merge with latest `main` branch - , Android build is broken (cannot run on Android device, getting a build error).

**Comment by kueda** [comment]
> Something to note - while trying to merge with latest main branch - , Android build is broken (cannot run on Android device, getting a build error).

I tried merging and was able to build on both but sharing won't work on iOS (obs edit never opens). Fun. I'll spend a little more time banging my head against it.

**Comment by kueda** [comment]
@budowski I made https://github.com/inaturalist/iNaturalistReactNative/tree/364-sharing-photos-to-inat-kueda, which builds on both platforms and works for me in Android, but in iOS (on device and in simulator), when I share a photo, iNat opens but it just lands on My Observations and the `handleShare` function never gets called. Can you spot anything in the iOS config there that looks off to you?

**Review by kueda — APPROVED** [review]

**Pull request merged**